browser/themes/brave_theme_service_browsertest.cc

```diff
@@ -305,36 +305,63 @@ IN_PROC_BROWSER_TEST_F(BraveThemeServiceTest, MAYBE_DarkModeChangeByRegTest) {
                                  &apps_use_light_theme);
   const bool initial_dark_mode = apps_use_light_theme == 0;
 
-  // Toggle dark mode and check get notification for default type (same as...).
+  // Set dark mode to "Same as Windows". In this mode we want to be receiving
+  // system notifications of the dark mode changes.
   dark_mode::SetBraveDarkModeType(
       dark_mode::BraveDarkModeType::BRAVE_DARK_MODE_TYPE_DEFAULT);
 
-  apps_use_light_theme = !initial_dark_mode ? 0 : 1;
-  hkcu_themes_regkey.WriteValue(L"AppsUseLightTheme", apps_use_light_theme);
-
-  TestNativeThemeObserver native_theme_observer_for_default;
-  EXPECT_CALL(
-      native_theme_observer_for_default,
-      OnNativeThemeUpdated(ui::NativeTheme::GetInstanceForNativeUi())).Times(1);
-  ui::NativeTheme::GetInstanceForNativeUi()->AddObserver(
-      &native_theme_observer_for_default);
-
-  // Toggle dark mode and |native_theme_observer_for_light| will not get
-  // notification for light type.
-  dark_mode::SetBraveDarkModeType(
-      dark_mode::BraveDarkModeType::BRAVE_DARK_MODE_TYPE_LIGHT);
-
-  TestNativeThemeObserver native_theme_observer_for_light;
-  EXPECT_CALL(
-      native_theme_observer_for_light,
-      OnNativeThemeUpdated(ui::NativeTheme::GetInstanceForNativeUi())).Times(0);
-  ui::NativeTheme::GetInstanceForNativeUi()->AddObserver(
-      &native_theme_observer_for_light);
+  {
+    // Set up theme observer and toggle the system dark mode by writing to the
+    // registry. We should get 2 notifications:
+    //    1 for dark mode change + 1 for reduced transparency.
+    // We get notifications for both because they are watching the same registry
+    // key.
+    TestNativeThemeObserver native_theme_observer_for_default;
+    ui::NativeTheme::GetInstanceForNativeUi()->AddObserver(
+        &native_theme_observer_for_default);
+
+    EXPECT_CALL(native_theme_observer_for_default,
+                OnNativeThemeUpdated(ui::NativeTheme::GetInstanceForNativeUi()))
+        .Times(2);
+    apps_use_light_theme = !initial_dark_mode ? 0 : 1;
+    hkcu_themes_regkey.WriteValue(L"AppsUseLightTheme", apps_use_light_theme);
+
+    // Timeout is used to let notifications trickle in.
+    RunLoopRunWithTimeout(base::Milliseconds(500));
+
+    // Toggling dark mode to light should result in only one notification since
+    // we aren't touching the registry.
+    EXPECT_CALL(native_theme_observer_for_default,
+                OnNativeThemeUpdated(ui::NativeTheme::GetInstanceForNativeUi()))
```

Comment:
This one time call is from reduced transparency?

Reply:
No, this is from our own code calling the notification directly.

Reply:
ahaaaa.. I just remembered. It's my code :) Thanks!